include/s3select_functions.h

```diff
@@ -1104,6 +1104,21 @@ struct _fn_diff_year_timestamp : public base_date_diff
     param_validation(args);
 
     int64_t yr = ptime2.date().year() - ptime1.date().year();
+
```

Comment:
please test the following
select date_diff(year, to_timestamp("2009-01-01T20:56:06.234Z"), to_timestamp("2008-12-31T19:30:05.234Z"))

Reply:
please review the time_duration api's https://www.boost.org/doc/libs/1_31_0/libs/date_time/doc/class_time_duration.html

it handle all time difference perspectives.

time_duration td = ptime2 - ptime1;
 td.hours() --> total amount of hours, in non-leap year its 8760 (in leap 8784)
 td.total_seconds()  --> *same(in seconds)*

the time_duration is accurate measure utility for time-diff calculation

Reply:
please try the following (and compare to AWS)
date_diff(year, to_timestamp('2010-12-31T00:29:04.234Z'), to_timestamp('2009-12-31T00:00:05.234Z')) 

the following provide correct calculation (and its sensitive to time and leap year)
date_diff(**day**, to_timestamp('2010-01-01T00:29:04.234Z'), to_timestamp('2009-01-01T00:30:05.234Z'))

(maybe it's possible to reuse)

Reply:
please validate the following query
date_diff(year, to_timestamp('2010-12-31T23:29:04.234Z'), to_timestamp('2008-12-31T00:00:05.234Z'))